packages/amplify-cli-core/src/amplify-node-pkg-detector/amplify-nodejs-detector.ts

```diff
@@ -0,0 +1,109 @@
+import * as fs from 'fs-extra';
+import _ from 'lodash';
+import * as path from 'path';
+import { AmplifyError } from '../errors/amplify-error';
+import { AmplifyFault } from '../errors/amplify-fault';
+import { getPackageManager, PackageManager } from '../utils/packageManager';
+import { Lockfile, LockfileParser } from './lock-file-interface';
+import { PackageJson } from './lock-file-types';
+import { LockFileParserFactory } from './parser-factory';
+
+/**
+ * return type of detectAffectedDirectDependencies
+ */
+type DetectedDependencies = {
+  packageName?: string;
+  dependentPackage?:{
+    name: string,
+    version: string,
+  }
+}
+
+/**
+ * props required by amplify detector
+ */
+export type AmplifyNodePkgDetectorProps = {
+    projectRoot: string,
+    dependencyToSearch: string,
+  }
+
+/**
+ * Amplify Node lock file dependency detector
+ * detects packages depending on pkg passed to Engine
+ * by parsing lock files
+ */
+export class AmplifyNodePkgDetector {
+     private readonly packageManager: PackageManager;
+     private readonly dependencyToSearch: string;
+     private readonly pkgJsonObj: PackageJson;
+     private readonly lockFileContents: string;
+     private readonly lockFileParser: LockfileParser;
+
+     constructor(amplifyDetectorProps: AmplifyNodePkgDetectorProps) {
+       const packageManager = getPackageManager(amplifyDetectorProps.projectRoot);
+       if (packageManager === null) {
+         throw new AmplifyError('MissingOverridesInstallationRequirementsError', {
+           message: 'No package manager found.',
+           resolution: 'Install npm or yarn to compile overrides for this project.',
+         });
+       }
+       this.packageManager = packageManager;
+       this.pkgJsonObj = this.parsePkgJson(amplifyDetectorProps.projectRoot);
+       this.dependencyToSearch = amplifyDetectorProps.dependencyToSearch;
+       this.lockFileContents = this.getFileContent(amplifyDetectorProps.projectRoot);
+       this.lockFileParser = LockFileParserFactory.getLockFileParser(this.packageManager.packageManager);
+     }
+
+     /**
+      * parses lock file
+      */
+     parseLockFile():Lockfile {
+       return this.lockFileParser.parseLockFile(this.lockFileContents);
+     }
+
+     /**
+    * parses package.json project files
+    */
+      private parsePkgJson = (projectRoot: string): PackageJson => {
+        const pkgJsonFullPath = path.resolve(projectRoot, 'package.json');
+        return <PackageJson>JSON.parse(fs.readFileSync(pkgJsonFullPath, 'utf-8'));
+      }
+
+     /**
+      * get file content as string
+      */
+     private getFileContent = (projectRoot: string) : string => {
+       const lockFileFullPath = path.resolve(projectRoot, this.packageManager.lockFile);
+       if (!fs.existsSync(lockFileFullPath)) {
+         throw new AmplifyFault('FileNotFoundFault', {
```

Comment:
```suggestion
         throw new AmplifyFault('LockFileNotFoundFault', {
```

Should this be error or fault?